Compiles and runs a test binary for testing cflags which fails in cross builds.
A check for cflags is not necessary anyway, so remove it completely.

--- a/Build.PL	2025-01-09 17:25:50.000000000 +0100
+++ b/Build.PL	2025-05-01 15:32:22.200963362 +0200
@@ -21,22 +21,6 @@
 
 my $cc = ExtUtils::CChecker->new( quiet => 1 );
 
-$cc->try_find_cflags_for(
-   cflags => [
-      # Most systems will need no extra cflags
-      [],
-      # HPUX may need to be told +std=gnu in order to accept anon inner unions
-      ( $^O eq "hpux" ) ? [ "+std=gnu" ] : (),
-   ],
-   source => <<'EOF'
-struct Named {
-   union { int a, b; };
-   int c;
-};
-int main(void) { struct Named n; n.a = 0; return n.a; }
-EOF
-) or die "OS unsupported - C compiler does not support anonymous inner unions\n";
-
 my $build = Module::Build::with::XSTests->new(
    module_name => 'XS::Parse::Keyword',
    requires => {
